capstone/src/main/webapp/js/demographicalGraphs.js

```diff
@@ -0,0 +1,184 @@
+ * Google Charts API and JSON object consisting of college demographical data.
+ */
+function graphDemographics() {
+  const admissionRateOptions = {
+    title: 'Admission Rate of Colleges',
+  };
+  const avgSatOptions = {
+    title: 'Average SAT Score of Colleges',
+  };
+  const numUndergradsOptions = {
+    title: 'Number of Undergraduate Students in Colleges',
+  };
+  const genderRatioOptions = {
+    title: 'Female/Male Ratio in Colleges',
+  };
+  const admissionRateChart =
+    new google.visualization.ColumnChart(
+        document.getElementById('admission-rate-data'),
+    );
+  const avgSatChart =
+    new google.visualization.ColumnChart(
+        document.getElementById('avg-sat-data'),
+    );
+  const numUndergradsChart =
+    new google.visualization.ColumnChart(
+        document.getElementById('num-undergrads-data'),
+    );
+  const genderRatioChart =
+    new google.visualization.ColumnChart(
+        document.getElementById('gender-ratio-data'),
+    );
+  deserializeGenderRatioData().then(
+      (data) => genderRatioChart.draw(data, genderRatioOptions),
+  );
+  deserializeAdmissionData().then(
+      (data) => admissionRateChart.draw(data, admissionRateOptions),
+  );
+  deserializeAvgSatData().then(
+      (data) => avgSatChart.draw(data, avgSatOptions),
+  );
+  deserializeNumUndergradsData().then(
+      (data) => numUndergradsChart.draw(data, numUndergradsOptions),
+  );
+}
+
+/**
+ * Helper function to fetch elements from JSON servlet.
+ *
+ * @return {google.visualization.DataTable} data the DataTable corresponding to
+ *     the admission rate data
+ */
+async function deserializeAdmissionData() {
+  const data = new google.visualization.DataTable();
+  await fetch('/viz/stats')
+      .then((response) => response.json())
+      .then(
+          (admissionInfo) =>
+            populateAdmissionRateDataTable(data, admissionInfo),
+      );
+  return data;
+}
+
+/**
+ * Helper function to fetch elements from JSON servlet.
+ *
+ * @return {google.visualization.DataTable} data the DataTable corresponding to
+ *     the average SAT score data
+ */
+async function deserializeAvgSatData() {
+  const data = new google.visualization.DataTable();
+  await fetch('/viz/stats')
+      .then((response) => response.json())
+      .then((avgSatInfo) => populateAvgSatDataTable(data, avgSatInfo));
+  return data;
+}
+
+/**
+ * Helper function to fetch elements from JSON servlet.
+ *
+ * @return {google.visualization.DataTable} data the DataTable corresponding to
+ *     the data on the number of undergraduate students per college
+ */
+async function deserializeNumUndergradsData() {
+  const data = new google.visualization.DataTable();
+  await fetch('/viz/stats')
+      .then((response) => response.json())
+      .then(
+          (numUndergradsInfo) =>
+            populateNumUndergradsDataTable(data, numUndergradsInfo),
+      );
+  return data;
+}
```

Comment:
This is going to result in a REST call for every graph. Is there a way we can consolidate those network calls to save resources?

I'm fine to address this in a follow up...

Reply:
Ah yea I was planning on addressing this later. I think what I'll try is fetching and passing a parameter consisting of the JSON object. Then not only will network calls be reduced, but all the populate functions can run in parallel.

Reply:
+1 to reducing network round trips.